### PR TITLE
Review payment identifies the recipient

### PR DESCRIPTION
**Pull request by mlsteele**
`reviewPaymentLocal` identifies the recipient. There are 3 outcomes.
1. The identify succeeds, the caller is notified, and the call exits. ReadyToSend
2. If an identify ever succeeds or the recipient is tracked or untracked by this device then the caller is notified just like 1.
3. The identify errors out, the caller gets a banner and then hangs out forever. If 2 ever happens that's great.

### CODE REVIEW

File: go/engine/identify2_with_uid.go [hunk 1]
```diff
@@ -867,6 +871,10 @@ func (e *Identify2WithUID) runIdentifyUI(m libkb.MetaContext) (err error) {
 		_, err = e.state.Result().GetErrorLax()
 	}
 
+	if outcome.IsOK() {
```
Comment (mlsteele) [inline go/engine/identify2_with_uid.go]: I don't really understand what's behind this. Why `TrackStatus_NEW_FAIL_PROOFS` isn't considered IsOk. But I tried a bunch of stuff and this seems like the behavior that is desired. The ideal is to find a condition that means that the state of their identity satisfies your tracking of them, or that you don't track them.

File: go/stellar/build.go [hunk 2]
```diff
@@ -386,6 +447,67 @@ func ReviewPaymentLocal(mctx libkb.MetaContext, stellarUI stellar1.UiInterface,
...
+	}
+
+	mctx.CDebugf("identifyForReview(%v)", assertion)
+	reason := fmt.Sprintf("Identify transaction recipient: %s", assertion)
+	eng := engine.NewResolveThenIdentify2(mctx.G(), &keybase1.Identify2Arg{
+		UserAssertion:         assertion,
+		CanSuppressUI:         true,
+		NoErrorOnTrackFailure: true, // take heed
+		Reason:                keybase1.IdentifyReason{Reason: reason},
+		IdentifyBehavior:      keybase1.TLFIdentifyBehavior_RESOLVE_AND_CHECK,
```
Comment (mlsteele) [inline go/stellar/build.go]: There are so many to choose from. Is this a good one? I just need track breaks to be part of the result, not error. And no tracker popups should pop up. This seems to unfortunately disable the cache too. @maxtaco

File: go/stellar/stellar.go [hunk 3]
```diff
@@ -515,6 +515,10 @@ func SendPaymentGUI(m libkb.MetaContext, walletState *WalletState, sendArg SendP
 func sendPayment(m libkb.MetaContext, walletState *WalletState, sendArg SendPaymentArg, isCLI bool) (res SendPaymentResult, err error) {
 	defer m.CTraceTimed("Stellar.SendPayment", func() error { return err })()
 
+	if err = m.Ctx().Err(); err != nil {
+		return res, err
+	}
```
Comment (patrickxb) [inline go/stellar/stellar.go]: what's this block for?

File: go/stellar/build.go (hunk 2 above)
Comment (patrickxb) [reply go/stellar/build.go]: I think this is good?

File: go/stellar/stellar.go (hunk 3 above)
Comment (mlsteele) [reply go/stellar/stellar.go]: It's just a friendly yield. When context's get canceled sometimes it's nice not to have them log a huge tree before realizing they've been canceled at the bottom of the callstack. But I don't remember why this one in particular and it doesn't seem likely for this to be canceled, so I'll remove it.

**Comment by mlsteele** [comment]
This one can wait until after code 🍟

**Comment by mlsteele** [comment]
nah

**Pull request merged**